frontend/src/libs/components/button/button.tsx

```diff
@@ -1,15 +1,27 @@
+import { type MouseEvent, type ReactNode } from 'react';
+
 import { ButtonType } from '~/libs/enums/enums.js';
 import { getValidClassNames } from '~/libs/helpers/helpers.js';
 import { type ValueOf } from '~/libs/types/types.js';
 
+import { Loader } from '../components.js';
 import styles from './styles.module.scss';
 
+type ButtonVariant = 'primary' | 'outlined' | 'text';
+
+type ButtonSize = 'medium' | 'small';
+
 type Properties = {
+  variant?: ButtonVariant;
+  size?: ButtonSize;
   disabled?: boolean;
   label: React.ReactNode;
   type?: ValueOf<typeof ButtonType>;
   name?: string;
   className?: string;
+  fullWidth?: boolean;
+  loading?: boolean;
```

Comment:
And not sure we need loadingLabel
```suggestion
  hasFullWidth?: boolean;
  isLoading?: boolean;
```